src/notebooks/outputs/tracebackFormatter.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable } from 'inversify';
+import { NotebookCell } from 'vscode';
+import { ITracebackFormatter } from '../../kernels/types';
+import { JupyterNotebookView } from '../constants';
+
+@injectable()
+export class NotebookTracebackFormatter implements ITracebackFormatter {
+    public format(cell: NotebookCell, traceback: string[]): string[] {
+        if (cell.notebook.notebookType !== JupyterNotebookView) {
+            return traceback;
```

Comment:
@rchiodo I found that the code for IW and Notebooks share very little similarities.
The only similarity is the regex stuff, which I plan on moving into a helper function or the like into `src/kernels/outputs/tracebackFormatter.ts` (which would be common to both formatters)

Basically notebook traceback formatters don't need the cell has providers at all.
& I'd like to make that very clear with the code (by having two seprate classes).
Also the cell hash provider is very specific to IW, hence that code only lives in the IW folder.

Reply:
Sounds good to me. I wasn't sure if that was the case but if it is all the better.

Reply:
Updated the code, not yet tested, 
will add tests